include/exadg/fluid_structure_interaction_precice/driver_fluid.h

```diff
@@ -0,0 +1,593 @@
+        "Fluid-Mesh-read",
+        "Velocity");
+      // TODO: Change to Velocity
+
+      VectorType initial_stress;
+      fluid_operator->initialize_vector_velocity(initial_stress);
+      initial_stress = 0;
+      this->precice->initialize_precice(initial_stress);
+    }
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+
+    // fluid
+    {
+      // setup time integrator before calling setup_solvers (this is necessary since the setup
+      // of the solvers depends on quantities such as the time_step_size or gamma0!!!)
+      AssertThrow(this->application->get_parameters_fluid().solver_type ==
+                    IncNS::SolverType::Unsteady,
+                  ExcMessage("Invalid parameter in context of fluid-structure interaction."));
+
+      // initialize fluid_time_integrator
+      fluid_time_integrator =
+        IncNS::create_time_integrator<dim, Number>(fluid_operator,
+                                                   this->application->get_parameters_fluid(),
+                                                   this->mpi_comm,
+                                                   this->is_test,
+                                                   fluid_postprocessor);
+
+      fluid_time_integrator->setup(this->application->get_parameters_fluid().restarted_simulation);
+
+      fluid_operator->setup_solvers(
+        fluid_time_integrator->get_scaling_factor_time_derivative_term(),
+        fluid_time_integrator->get_velocity());
+    }
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+  }
+
+  void
+  solve() const final
+  {
+    Assert(this->application->get_parameters_fluid().adaptive_time_stepping == false,
+           ExcNotImplemented());
+
+    bool is_new_time_window = true;
+
+    dealii::Timer precice_timer;
+    // preCICE dictates when the time loop is finished
+    while(this->precice->is_coupling_ongoing())
+    {
+      // pre-solve
+      fluid_time_integrator->advance_one_timestep_pre_solve(is_new_time_window);
+
+      this->precice->save_current_state_if_required([&]() { /*TODO*/ });
+
+      // solve (using strongly-coupled partitioned scheme)
+      {
+        coupling_structure_to_ale();
+
+        // move the fluid mesh and update dependent data structures
+        solve_ale();
+
+        // update velocity boundary condition for fluid
+        coupling_structure_to_fluid();
+
+        // solve fluid problem
+        fluid_time_integrator->advance_one_timestep_partitioned_solve(is_new_time_window, true);
+
+        // compute and send stress to solid
+        coupling_fluid_to_structure();
+
+        precice_timer.restart();
+        this->precice->advance(fluid_time_integrator->get_time_step_size());
+        is_new_time_window = this->precice->is_time_window_complete();
```

Comment:
Does this involve a synchronisation step where the fluid and solid solvers wait for each other? Put differently, how is it ensured that the solvers for the fluid and solid problems do not drift away from each other in time, but compute one time step after the other in-sync?
Comments in the code in `DriverFluid` and `DriverSolid` would be helpful.